src/index.ts

```diff
@@ -1,104 +1,155 @@
+import packageJson from "../package.json" assert { type: "json" };
 
-import { VERSION } from './version';
+const API_URL = "https://api.zenrows.com/v1/";
 
-const API_URL = 'https://api.zenrows.com/v1/';
+type HttpMethods = "GET" | "POST" | "PUT";
 
 interface ClientConfig {
-    concurrency?: number;
-    retries?: number;
+  concurrency?: number;
+  retries?: number;
 }
-export interface ZenRowsConfig {
-	autoparse?: boolean;
-	css_extractor?: string;
-	js_render?: boolean;
-	premium_proxy?: boolean;
-	proxy_country?: string;
-	wait_for?: string;
-	wait?: number;
-	block_resources?: string;
-	window_width?: number;
-	window_height?: number;
-	device?: string;
-	original_status?: boolean;
-	custom_headers?: boolean;
-	[x: string]: unknown;
+export interface ZenrowsConfig {
+  autoparse?: boolean;
+  css_extractor?: string;
+  js_render?: boolean;
+  premium_proxy?: boolean;
+  proxy_country?: string;
+  wait_for?: string;
+  wait?: number;
+  block_resources?: string;
+  window_width?: number;
+  window_height?: number;
+  device?: string;
+  original_status?: boolean;
+  custom_headers?: boolean;
+  [x: string]: unknown;
 }
 
 interface Headers {
-    [x: string]: string;
+  [x: string]: string;
 }
 
-class ZenRows {
-    readonly apiKey: string;
-
-    readonly clientConfig: ClientConfig;
-
-    readonly queue;
-
-    constructor(apiKey: string, clientConfig: ClientConfig = {}) {
-        this.apiKey = apiKey;
-        this.clientConfig = clientConfig;
+export class ZenRows {
+  readonly apiKey: string;
+  readonly clientConfig: ClientConfig;
+  readonly queue;
+  readonly fetchWithRetry;
+
+  constructor(apiKey: string, clientConfig: ClientConfig = {}) {
+    this.apiKey = apiKey;
+    this.clientConfig = clientConfig;
+    const retries = this.clientConfig.retries ?? 0;
+
+    this.queue = fastq.promise(
+      this,
+      this.worker,
+      this.clientConfig.concurrency ?? 5,
+    );
+
+    this.fetchWithRetry = fetchRetry(fetch, {
+      retryDelay: (attempt) => 2 ** attempt * 1000,
+      // retryOn: [422, 503, 504],
+      retryOn: (attempt, error, response) => {
+        if (attempt > retries) {
+          return false;
+        }
 
-        this.queue = fastq.promise(this, this.worker, this.clientConfig.concurrency ?? 5);
+        if (
+          error !== null ||
+          response?.status === 422 ||
+          response?.status === 503 ||
+          response?.status === 504
+        ) {
+          return true;
+        }
 
-        this.applyRetries();
+        return false;
+      },
+    });
+  }
+
+  public get(
+    url: string,
+    config?: ZenrowsConfig,
+    { headers = {} }: { headers?: Headers } = {},
+  ): Promise<Response> {
+    return this.queue.push({ url, config, headers });
+  }
+
+  public post(
+    url: string,
+    config?: ZenrowsConfig,
+    { headers = {}, data = {} }: { headers?: Headers; data?: unknown } = {
+      headers: { "Content-Type": "application/x-www-form-urlencoded" },
```

Comment:
Is the default Content-Type needed now? Forced below as default in a different way.

Reply:
Yeah, can be removed as per my last change. Done as well.